src/org/spdx/html/LicenseTOCJSONFile.java

```diff
@@ -0,0 +1,150 @@
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+package org.spdx.html;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+
+import org.spdx.rdfparser.license.SpdxListedLicense;
+
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+
+/**
+ * This class holds a JSON file for a license table of contents
+ * @author Kyle E. Mitchell
+ *
+ */
+public class LicenseTOCJSONFile {
+	
+	public class ListedSpdxLicense {
+		private String reference;
+		private String refNumber;
+		private String licenseId;
+		private boolean osiApproved;
+		private String licenseName;
+		
+		public ListedSpdxLicense() {
+			reference = null;
+			refNumber = null;
+			licenseId = null;
+			osiApproved = false;
+			licenseName = null;
+		}
+		
+		public ListedSpdxLicense(String reference, String refNumber, 
+				String licenseId, boolean isOsiApproved, String licenseName) {
+			this.reference = reference;
+			this.refNumber = refNumber;
+			this.licenseId = licenseId;
+			this.osiApproved = osiApproved;
+			this.licenseName = licenseName;
+		}
+
+		public String getReference() {
+			return reference;
+		}
+
+		public void setReference(String reference) {
+			this.reference = reference;
+		}
+
+		public String getRefNumber() {
+			return refNumber;
+		}
+
+		public void setRefNumber(String refNumber) {
+			this.refNumber = refNumber;
+		}
+
+		public String getLicenseId() {
+			return licenseId;
+		}
+
+		public void setLicenseId(String licenseId) {
+			this.licenseId = licenseId;
+		}
+
+		public boolean getOsiApproved() {
+			return osiApproved;
+		}
+
+		public void setOsiApproved(boolean osiApproved) {
+			this.osiApproved = osiApproved;
+		}
+
+		public String getLicenseName() {
+			return licenseName;
+		}
+
+		public void setLicenseName(String licenseName) {
+			this.licenseName = licenseName;
+		}
+	}
+	
+	ArrayList<ListedSpdxLicense> listedLicenses = new ArrayList<ListedSpdxLicense>();
+	
+	private int currentRefNumber = 1;
+
+	String version;
+	String releaseDate;
+
+	public LicenseTOCJSONFile(String version, String releaseDate) {
+		this.version = version;
+		this.releaseDate = releaseDate;
+	}
+
+	public void addLicense(SpdxListedLicense license, String licHTMLReference) {
+		listedLicenses.add(new ListedSpdxLicense(licHTMLReference, String.valueOf(this.currentRefNumber), 
+				license.getLicenseId(), license.isOsiApproved(), license.getName()));
+		currentRefNumber++;
+	}
+
+	public void writeToFile(File jsonFile) throws IOException {
+		FileWriter writer = null;
+		if (!jsonFile.exists()) {
+			if (!jsonFile.createNewFile()) {
+				throw(new IOException("Can not create new file "+jsonFile.getName()));
+			}
+		}
+		try {
+			JSONObject jsonObject = new JSONObject();

```

Comment:
@sindresorhus, I take it this JSON format for the list of SPDX license identifiers would work fine. Please feel free to share any thoughts.